### PR TITLE
Basic hint text for password.

### PR DESCRIPTION
**Pull request by samaratrilling**
[ch 7048]
Screen Shot 2021-05-14 at 10 43 58 AM

Very open to suggestions on styling, as well as how to localize this.

### CODE REVIEW

**Comment by toolness** [comment]
Nice!  You're about to read some stuff that you may have already anticipated--I wrote them and then realized this PR is still a draft--sorry if that's the case!

The one thing I'd recommend is to add a comment above `AUTH_PASSWORD_VALIDATORS` in our `settings.py` file that says something along the lines of "make sure these are in-sync with the human-readable guidelines for password creation defined in `<filename>.tsx`.  That way if we ever decide to change the password requirement logic, we (hopefully) won't forget to change the human-readable explanation too.

In fact, just so the documentation never rots, it might be useful to factor out the password guidelines into their own separate file that any password fields import from.  Especially since we actually have a number of different places where we ask users for new passwords across all our products!

**Comment by samaratrilling** [comment]
Cool! I'm exploring factoring out the two password creation fields into their own component.

**Comment by samaratrilling** [comment]
This still needs Spanish translation for the new strings.

**Review by toolness — APPROVED** [review]
Nice, this looks really good!  Aside from providing helpful text to users, it also makes our codebase more DRY, which rocks.  Thanks also for mentioning the related `settings.py` and `create-password.tsx` files in comments.

File: frontend/lib/common-steps/create-password.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+} from "../forms/form-fields";
+import { li18n } from "../i18n-lingui";
+
+// See project/settings.py for the list of password validators.
+const PASSWORD_VALIDATATION_HINT = t`Must be at least 8 characters. Can't be all numbers.`;
+
+type CreatePasswordProps = {
+  passwordProps: BaseFormFieldProps<string>;
+  confirmPasswordProps: BaseFormFieldProps<string>;
+  renderPasswordLabel?: LabelRenderer;
```
[the remark this answers is not in the document]
Comment (samaratrilling) [reply frontend/lib/common-steps/create-password.tsx]: Eh, I'll take it out.

**Pull request merged**